core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1183,14 +1222,16 @@ public Builder setStatistics(long snapshotId, StatisticsFile statisticsFile) {
           "snapshotId does not match: %s vs %s",
           snapshotId,
           statisticsFile.snapshotId());
-      statisticsFiles.put(statisticsFile.snapshotId(), ImmutableList.of(statisticsFile));
```

Comment:
The interface supports returning multiple stats file for one snapshot. But was always overwriting instead of appending. So, fixed it.

Reply:
This is not correct. There should be only one stats file for a snapshot that contains all information. Allowing multiple stats files for a snapshot is going to lead to lazy implementations that don't merge the files and slower job planning.

Reply:
@rdblue, @findepi:  
It is just as per the previous existing interface in the builder. 
we had `Map<Long, List<StatisticsFile>> statisticsFiles` in builder. 
If we wanted one stats file per interface then no need to have list for map value?

Also is it possible that there will be one partition level stats (avro/parquet file) and there is one table level NDV file per snapshot. In this case, no need to merge them but can be used based on the need. So, still ok to have multiple stats per snapshot id?